File: it.unibo.boundaryWalk/app/src/main/java/it/unibo/annotatedExample/ISSActor.java
/*
ISSActor.java
 */
package it.unibo.annotatedExample;



//@ISSActorSpec( actorName = "actordefault" )
public abstract class ISSActor {
    protected String myname = "unknown";

    @ISSActorSpec
    public ISSActor( ){
        myname = UniboAnnotationUtil.getActorName( this.getClass() );
        UniboAnnotationUtil.initializeObject( this );   //calls a method of the user class
        System.out.println("ISSActor | create " + myname);
    }
/*
    @InitSpec
    public void show(){
        System.out.println( myname + " ISSActor show | myname= " + myname);
    }*/

    public abstract void show();
}

File: it.unibo.boundaryWalk/app/src/main/kotlin/it/unibo/annotExample/ISSActor.kt
/*
ISSActor.kt
 */
package it.unibo.annotExample

abstract class ISSActor @ISSActorSpec constructor() {
    protected var myname = "unknown"

    abstract fun show()

    init {
        myname = UniboAnnotationUtil.getActorName(this.javaClass)
        UniboAnnotationUtil.initializeObject(this) //calls a method of the user class
        println("ISSActor | create $myname")
    }
}